staging/src/k8s.io/client-go/rest/request.go

```diff
@@ -122,6 +122,12 @@ func NewRequest(client HTTPClient, verb string, baseURL *url.URL, versionedAPIPa
 	if baseURL != nil {
 		pathPrefix = path.Join(pathPrefix, baseURL.Path)
 	}
+	var c context.Context
+	if timeout > 0 {
+		c, _ = context.WithTimeout(context.Background(), timeout)
```

Comment:
can we ignore `cancel` function that is returned? not that I have ideas about when it should be called.

Reply:
Also problem with this approach is the deadline will start at the time of context is created, here? If that is the case, probably you can implement timeout in the request itself? then you also know when request is done and you can call cancel function.

Reply:
@mbohlool I added some comment below. Please take a look.